### ISSUE TITLE
Not able to refresh database after editing a form title

### ISSUE THREAD

**Issue by sudhan77** [plomino/Plomino#609]
I had a form 'InterestProfiler -Web'. I edited it to 'InterestProfiler-Web'. However, I am unable refresh the database after that with the following errors: Earlier, I had resolved such error by deleting the documents. But that does not work now.

You are here: Home / MitMit Jobs & Internships
Error
Form InterestProfiler -Web does not exist.
We’re sorry, but there seems to be an error…
Here is the full error message:

Display traceback as text

Traceback (innermost last):

Module ZPublisher.Publish, line 138, in publish
Module ZPublisher.mapply, line 77, in mapply
Module ZPublisher.Publish, line 48, in call_object
Module <string>, line 4, in refreshPortalCatalog
Module AccessControl.requestmethod, line 70, in _curried
Module Products.CMFPlomino.PlominoDesignManager, line 368, in refreshPortalCatalog
Module Products.CMFPlone.CatalogTool, line 353, in catalog_object
Module Products.ZCatalog.ZCatalog, line 476, in catalog_object
Module Products.ZCatalog.Catalog, line 334, in catalogObject
Module Products.ZCatalog.Catalog, line 284, in updateMetadata
Module Products.ZCatalog.Catalog, line 410, in recordify
Module Products.CMFPlomino.PlominoDocument, line 608, in Title
AttributeError: 'NoneType' object has no attribute 'getDynamicDocumentTitle'

**Comment by ebrehault** [comment]
When you change the form id, the documents previously existing are still have the old id stored in their `Form` item, that's why you get errors.
You need to write an agent to go over all the existing docs and replace their form:
```python
db = context.getParentDatabase()
for doc in db.getAllDocuments():
    if doc.Form == "InterestProfiler -Web":
        doc.setItem('Form', 'InterestProfiler-Web')
        doc.save()
```

**Comment by sudhan77** [comment]
Thanks Eric.
However, I find the quirks of Plomino very confusing. I have again got
stuck because I added a field called 'creator' in an existing form. My
intention is to control access to the document to the author of the
document. I am faced with a none type error again.

On Fri, Apr 3, 2015 at 2:33 PM, Eric BREHAULT <notifications@github.com>
wrote:

> Closed #609 <https://github.com/plomino/Plomino/issues/609>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/plomino/Plomino/issues/609#event-272526818>.
>



-- 
Sudhanshu

**Comment by ebrehault** [comment]
`creator` is a Plone reserved id.
A good practice in Plomino is to use a prefix for your fields, for instance `interestprofile_creator`.

Regarding access control, you should read the documentation about `Plomino_Readers` and `Plomino_Authors` fields.
